src/node/internal/crypto_keys.ts

```diff
@@ -130,7 +130,7 @@ export abstract class KeyObject {
       if (typeof opts.passphrase === 'string') {
         opts.passphrase = Buffer.from(opts.passphrase, opts.encoding);
       }
-      if (!isUint8Array(opts.passphrase)) {
```

Comment:
We don't check that if a cipher or passphrase is specified the other one is specified too. This should be fine if just a passphrase is specified (which is ignored), but the OpenSSL documentation states that if there is a cipher without passphrase functions like `PEM_write_bio_PKCS8PrivateKey` will use a default callback to try to get a passphrase – not sure if that would affect us but it's probably better to provide an error here if needed.

Reply:
This is (should be) handled at the c++ level and follows Node.js' guidance. For instance, if a zero-length password is passed we set it to nullptr to specifically avoid the blocking password callback from being invoked.

Reply:
As far as I can see OpenSSL uses the callback if the passphrase is null:

https://www.openssl.org/docs/man3.1/man3/PEM_write_bio_RSAPrivateKey.html
> The cb argument is the callback to use when querying for the pass phrase used for encrypted PEM structures (normally only private keys).
For the PEM write routines if the kstr parameter is not NULL then klen bytes at kstr are used as the passphrase and cb is ignored.
If the cb parameters is set to NULL and the u parameter is not NULL then the u parameter is interpreted as a NUL terminated string to use as the passphrase. If both cb and u are NULL then the default callback routine is used which will typically prompt for the passphrase on the current terminal with echoing turned off.

Reply:
You resolved this issue, but when using `jwk1_ko` from the keys tests the following code triggers the issue:
```
jwk1_ko.export({
  type: "pkcs8",
  format: "pem",
  cipher: "aes-128-cbc",
  passphrase: ""
});
```

Note that unlike OpenSSL BoringSSL just returns an error if no passphrase is available instead of querying the command line, but we still need to handle this case. Node avoids the issue by returning an error if cipher is defined and passphrase is undefined and by setting passphrase to an arbitrary nonzero pointer and setting pass_len to zero if a zero-length passphrase was provided.

Reply:
ah, forgot to include that fix in the commit. Had it locally but didn't add it to the set.